frontend/app/utils/date-utils.ts

```diff
@@ -68,3 +103,27 @@ export function getAgeFromDateString(date: string) {
   invariant(isPast(parsedDate), `date must be in past [${date}]`);
   return differenceInYears(new Date(), parsedDate);
 }
+
+/**
+ * Checks if a string is a validad date. Else returns false.
+ * @param dateParts String[3] with the corresponding date components
+ * @param pattern Format to check against.
+ * @returns An object containing the parsed components (year, month, day). Returns { day: -1, month: -1, year: -1 } if the date does not exist or is invalid.
+ */
+function parseDateValue(dateParts: string[], pattern: string): { day: number; month: number; year: number } {
```

Comment:
This can already be acheive with date-fns. It's an established pattern without the app.

```js
const dateString = '1985-01-10';
const parsedDate = parse('dateString', 'yyyy-MM-dd', new Date());
const result = isValid(parsedDate); // validate date after parse
//=> true
```

https://date-fns.org/v3.6.0/docs/parse
https://date-fns.org/v3.6.0/docs/isValid

Reply:
It's not needed and I recommend we remove it and stick to the establish patterns. This will cause confusions.